src/library/scala/sys/process/ProcessImpl.scala

```diff
@@ -32,14 +35,14 @@ private[process] trait ProcessImpl {
   }
   private[process] object Future {
     def apply[T](f: => T): (Thread, () => T) = {
-      val result = new SyncVar[Either[Throwable, T]]
+      val result = Promise[Either[Throwable, T]]()
       def run(): Unit =
-        try result.put(Right(f))
-        catch { case e: Exception => result.put(Left(e)) }
+        try result.trySuccess(Right(f))
+        catch { case e: Exception => result.trySuccess(Left(e)) }
```

Comment:
If this is not Exception, won't line 45 block forever?

Reply:
Do will.

Reply:
:+1:

Reply:
I changed it to `case e:Throwable`